indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -348,6 +348,44 @@ public boolean registerUpgradedPendingSegmentOnSupervisor(
     return false;
   }
 
+  public boolean changeTaskCountSupervisor(String id, Integer taskCount)
+  {
+    Preconditions.checkState(started, "SupervisorManager not started");
+    Preconditions.checkNotNull(id, "id");
+    Preconditions.checkState(
+        taskCount != null && taskCount >= 0,
+        "taskCount should be non null positive value"
+    );
+
+    synchronized (lock) {
+      Preconditions.checkState(started, "SupervisorManager not started");
+      return changeTaskCountSupervisorInternal(id, taskCount);
+    }
+  }
+
+  /**
+   * Scale a supervisor with a given id.
+   * <p/>
+   * Caller should have acquired [lock] before invoking this method to avoid contention with other threads that may be
+   * starting, stopping, suspending and resuming supervisors.
+   *
+   * @return true if a supervisor was scaled, false if there was no supervisor with this id.
+   */
+  private boolean changeTaskCountSupervisorInternal(String id, Integer taskCount)
+  {
+    Pair<Supervisor, SupervisorSpec> pair = supervisors.get(id);
+    if (pair == null) {
+      return false;
+    }
+
+    try {
+      return pair.lhs.changeTaskCount(taskCount);
```

Comment:
how does this work with running tasks? would the next set of tasks spawn with the new task count?

Reply:
yeah.

Reply:
i wonder if we should return a 202 here instead since its not a immediate action. might make sense to keep this separate too then re: @suneet-s's comment. since the regular supervisor post expects changes to take place immediately.

Reply:
modified to call post instead.